shared/actions/provision.tsx

```diff
@@ -26,80 +27,82 @@ type ValidCallback =
   | 'keybase.1.provisionUi.ProvisioneeSuccess'
   | 'keybase.1.provisionUi.ProvisionerSuccess'
   | 'keybase.1.provisionUi.chooseDevice'
+  | 'keybase.1.provisionUi.chooseDeviceType'
   | 'keybase.1.provisionUi.chooseGPGMethod'
   | 'keybase.1.provisionUi.switchToGPGSignOK'
   | 'keybase.1.secretUi.getPassphrase'
 
 const ignoreCallback = (_: any) => {}
 
+type CustomParam<T extends ValidCallback> = RPCTypes.MessageTypes[T]['inParam']
+type CustomResp<T extends ValidCallback> = {
+  error: RPCTypes.IncomingErrorCallback
+  result: (res: RPCTypes.MessageTypes[T]['outParam']) => void
+}
+
 // The provisioning flow is very stateful so we use a class to handle bookkeeping
 // We only allow one manager to be alive at a time
 // Can be made for a regular provision or if we're adding a device
 class ProvisioningManager {
-  static singleton: ProvisioningManager | null = null
+  static singleton: ProvisioningManager | undefined
   static getSingleton = (): ProvisioningManager => {
     if (!ProvisioningManager.singleton) {
       throw new Error('No ProvisioningManager')
     }
     return ProvisioningManager.singleton
   }
-  _stashedResponse: any = null
```

Comment:
before we kept one value and used a key. this meant it's impossible to type correctly. instead we have an object that maps correctly